vllm/model_executor/model_loader/loader.py

```diff
@@ -704,13 +711,54 @@ def _get_quantized_weights_iterator(
             model_name_or_path, revision)
 
         quant_state_dict = {}
-        if use_safetensors:
-            weight_iterator = safetensors_weights_iterator(hf_weights_files)
-        else:
-            weight_iterator = pt_weights_iterator(hf_weights_files)
 
-        def generator():
+        def prequant_generator() -> Generator:
+            #first iterate over all quant state weights
+            weight_iterator = self._hf_weight_iter(hf_weights_files,
+                                                   use_safetensors)
+            temp_state_dict = {}
             for weight_name, weight_tensor in weight_iterator:
+                if weight_name.endswith(".weight"):
+                    continue
+                temp_state_dict[weight_name] = weight_tensor
+
+            #closure to parse quant_state for each prequant weight
+            def _parse_quant_state(param_name: str,
+                                   temp_state_dict: Dict) -> QuantState:
+                quant_state = {}
+                for k in temp_state_dict:
+                    if param_name + "." in k:
+                        quant_state[k] = temp_state_dict[k]
+                #bitsandbytes library requires
+                #weight.quant_state.bitsandbytes__nf4 in CPU
+                quant_state[param_name +
+                            ".quant_state.bitsandbytes__nf4"] = quant_state[
+                                param_name +
+                                ".quant_state.bitsandbytes__nf4"].cpu().data
+                return QuantState.from_dict(quant_state, device="cuda")
+
+            #second iterate over all prequant and normal weights
+            #pre quant weights would have a quant_state
+            for weight_name, weight_tensor in self._hf_weight_iter(
+                    hf_weights_files, use_safetensors):
+                #filter all weights whose suffix is not ".weight"
+                if not weight_name.endswith(".weight"):
+                    continue
+                #.quant_state.bitsandbytes__nf4 is not supported
```

Comment:
What isn't supported about it? It seems like no exception is being thrown

Reply:
a typo here, it should be "only quant_state.bitsandbytes__nf4 is supported".  other libraries such as hf transformer supports quant_state.bitsandbytes__fp4.